### PR TITLE
Test macos installation

### PR DESCRIPTION
**Pull request by lxn2**
This PR adds logic to test @sandeep-krishnamurthy's python installation guide for MacOS CPU on Travis.

We test the installation in a clean environment via pip, virtualenv, and from source.

### CODE REVIEW

**Comment by mli** [comment]
can you please add a little bit context here? do you want to test all the installation on travis? how many time will it cost? my concern is that our travis testing is pretty slow so we'd like not to add heavy tasks to it.

**Comment by lxn2** [comment]
@mli yes, sure.

There are two tasks that I've added to Travis as shown in the .travis.yml file:

1. installation_packaged_test

- Tests instructions for installing MXNet via either **1) virtualenv 2) pip**
- Runs a simple Python sanity test (create NDArray)
- Takes 2 minutes

2. installation_source_test

- Tests instructions for installing MXNet from **source**
- Runs a simple Python sanity test (create NDArray)
- Takes 16 minutes
- Needed to be put in a second task because it needs a clean environment

A successful run can be viewed here: https://travis-ci.org/lxn2/mxnet/builds/231756599

We did consider that it might cause Travis runs to be slower too, but testing installation instructions was high priority for v0.10 release and that's why we committed time to it.

**Comment by lxn2** [comment]
Perhaps I can add to the following file to check whether the install.md guide has changed. If it has changed, we run the task. If not, we skip? Although I notice we've disabled this check...
https://github.com/dmlc/mxnet/blob/master/tests/travis/is_core_changed.sh

**Comment by bhavinthaker** [comment]
@lxn2 @mli @sandeep-krishnamurthy I vote for making the macOS build automation as a nightly test and not something that runs for every code submission.

**Comment by lxn2** [comment]
Closing temporarily.

**Comment by lxn2** [comment]
I added logic to run installation tasks only if the diff between this branch and the mxnet's master branch contains install.md.

Note the installation_source_test task fails in Travis because my changes to `install-mxnet-osx-python.sh` needs to be merged.

Also @mli the Jenkins server couldn't clone this repo for some reason...can you help retrigger the build?

**Comment by mli** [comment]
i don't know jenkins failed to clone this pr, maybe due to there is too many commits. but it seems to be fixed now

**Comment by lxn2** [comment]
Yes I think it's a known issue with git pull request builder.

File: .travis.yml [hunk 1]
```diff
@@ -16,8 +16,10 @@ env:
   # run tests/cpp
   - TASK=cpp_test
   # run tests/python
-  - TASK=python_test
```
Comment (piiswrong) [inline .travis.yml]: ？

**Comment by piiswrong** [comment]
Why are we testing pip install here? Shouldn't it be tested in mxnet-distro? That's the place that's creating the pip packages.

File: .travis.yml (hunk 1 above)
Comment (lxn2) [inline .travis.yml]: @mli asked me to remove Python and R tests. Please let me know if you want to keep it in or not.

**Comment by lxn2** [comment]
@piiswrong This tests the _installation instructions_ to make sure that the guide works. The testing in mxnet-distro needs to happen too, but that is a separate in-progress item.

File: .travis.yml (hunk 1 above)
Comment (piiswrong) [inline .travis.yml]: I think its better to keep python tests on mac until we have our own mac servers.

**Comment by piiswrong** [comment]
pip install depends on the pip package which is not in this repo. Putting it here can cause test fails without changes in mxnet repo. These tests should be moved into mxnet-distro

source install tests can be here.

**Comment by lxn2** [comment]
I advocate to push this in because
- this should on run if there is a change to install.md installation guide
- we're in progress of moving mxnet-distro into mxnet anyway @szha

**Comment by szha** [comment]
I've added the tests in mxnet-distro for the immediate release.

Eric has a point that having external dependencies for tests is bad and unless we mock the dependencies there isn't a way around it. Also, adding a test for `pip install mxnet` doesn't help much anyways. Ly, see if you could find a way to mock the invocation of commands, or come up with a better way, or exclude the pip command from doc tests.

**Pull request merged**